### PR TITLE
Angular integration for Anchor and Anchor Button

### PR DESCRIPTION
**Pull request by m-akinc**
# Pull Request

## 🤨 Rationale

Anchor and Anchor Button need Angular integration.

## 👩‍💻 Implementation

- Created directives
- Added to example app
- Updated README

## 🧪 Testing

Automated tests created

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

**Review by jattasNI — APPROVED** [review]

File: angular-workspace/projects/ni/nimble-angular/src/directives/patterns/link/nimble-link-base.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import { ElementRef, Injectable, Input, Renderer2 } from '@angular/core';
+import type { AnchorBase } from '@ni/nimble-components/dist/esm/anchor-base';
+
+/**
+ * Base class for directives providing link attributes.
+ */
+@Injectable()
+export class NimbleLinkBase<T extends AnchorBase> {
```
Comment (rajsite) [inline angular-workspace/projects/ni/nimble-angular/src/directives/patterns/link/nimble-link-base.ts:8]: Should match the base class style of icon-base. Class name should be NimbleAnchorBaseDirective and file should be 
/directives/anchor-base/nimble-anchor-base.directive.ts (aligns with the /directives/icon-base conventions).

Comment (rajsite) [inline angular-workspace/projects/ni/nimble-angular/src/directives/patterns/link/nimble-link-base.ts:8]: I don't think this class needs to be generic as long as the other anchor directives extend this class

Comment (m-akinc) [reply angular-workspace/projects/ni/nimble-angular/src/directives/patterns/link/nimble-link-base.ts:8]: Without that generic type constraint, I get errors trying to reference the properties on `nativeElement` in the getters, e.g. `this.elementRef.nativeElement.href`. Also, isn't it just better to enforce that everything extending this derives from AnchorBase?

Comment (m-akinc) [reply angular-workspace/projects/ni/nimble-angular/src/directives/patterns/link/nimble-link-base.ts:8]: Done

Comment (rajsite) [reply angular-workspace/projects/ni/nimble-angular/src/directives/patterns/link/nimble-link-base.ts]: makes sense that it's needed for the nativeElement refs 👍

**Review by rajsite — APPROVED** [review]

**Pull request merged**